doc/event-alarm-framework/event-alarm-framework.md

```diff
@@ -0,0 +1,1265 @@
+```
+
+The following is SONiC yang for alarms.
+```
+module: sonic-alarm
+  +--rw sonic-alarm
+     +--rw ALARM
+     |  +--rw ALARM_LIST* [id]
+     |     +--rw id                  uint64
+     |     +--rw resource?           string
+     |     +--rw text?               string
+     |     +--rw time-created?       event:timeticks64
+     |     +--rw type-id?            string
+     |     +--rw severity?           event:severity-type
+     |     +--rw acknowledged?       boolean
+     |     +--rw acknowledge-time?   event:timeticks64
+     +--rw ALARM_STATS
+        +--rw ALARM_STATS_LIST* [id]
+           +--rw id              enumeration
+           +--rw alarms?         uint64
+           +--rw critical?       uint64
+           +--rw major?          uint64
+           +--rw minor?          uint64
+           +--rw warning?        uint64
+           +--rw acknowledged?   uint64
+
+  rpcs:
+    +---x acknowledge-alarms
+    |  +---w input
+    |  |  +---w id*   string
+    |  +--ro output
+    |     +--ro status?          int32
+    |     +--ro status-detail?   string
+    +---x unacknowledge-alarms
+    |  +---w input
+    |  |  +---w id*   string
+    |  +--ro output
+    |     +--ro status?          int32
+    |     +--ro status-detail?   string
+    +---x show-alarms
+       +---w input
+       |  +---w (option)?
+       |     +--:(time)
+       |     |  +---w time
+       |     |     +---w begin?   yang-types:date-and-time
+       |     |     +---w end?     yang-types:date-and-time
+       |     +--:(last-interval)
+       |     |  +---w interval?   enumeration
+       |     +--:(severity)
+       |     |  +---w severity?   event:severity-type
+       |     +--:(id)
+       |        +---w id
+       |           +---w begin?   string
+       |           +---w end?     string
+       +--ro output
+          +--ro status?          int32
+          +--ro status-detail?   string
+          +--ro ALARM
+             +--ro ALARM_LIST* [id]
+                +--ro id                  uint64
+                +--ro resource?           string
+                +--ro text?               string
+                +--ro time-created?       event:timeticks64
+                +--ro type-id?            string
+                +--ro severity?           event:severity-type
+                +--ro acknowledged?       boolean
+                +--ro acknowledge-time?   event:timeticks64
+```
+
+Following is for sonic yang to support event profiles.
+```
+module: sonic-evprofile
+
+  rpcs:
+    +---x get-evprofile
+    |  +--ro output
+    |     +--ro file-name?   string
+    |     +--ro file-list*   string
+    +---x set-evprofile
+       +---w input
+       |  +---w file-name?   string
+       +--ro output
+          +--ro status?   string
+```
+
+openconfig alarms yang is defined at [here](https://github.com/openconfig/public/blob/master/release/models/system/openconfig-alarms.yang)
+
+### 3.3.2 CLI
+#### 3.3.2.1 Exec Commands
```

Comment:
As previously noted I think the CLI commands should be detailed described in an HLD.
I would like to see the command name, the parameters, the default behaviour and the output of the command.

My assumption is that these commands are provided as native SONiC CLI commands which are also to be shared in the command reference guide. it is recommended to have in the HLD the same format as in the command reference guide and later it can be simple copy/paste.

Reply:
I am confused. Didnt we add description/example output of commands in section 3.2.2 ?

Will update them in the command ref guide format.